src/main/java/de/rabea/game/Board.java

```diff
@@ -10,23 +10,34 @@
 public class Board {
 
     private Mark[] cells;
+    private int dimension;
+    private int numberOfCells;
 
+    public Board(Mark... gameState) {
+        this.dimension = (int) Math.sqrt(gameState.length);
+        this.numberOfCells = gameState.length;
+        this.cells = gameState;
+    }
 
-    public Board() {
-       this(new Mark[] {EMPTY, EMPTY, EMPTY,
-                EMPTY, EMPTY, EMPTY,
-                EMPTY, EMPTY, EMPTY});
+    public Board(int dimension) {
+        this.dimension = dimension;
```

Comment:
If you have more than constructor, then they should each other. It's called a 'telescope constructor'.
This should also drop the number instance variables you need.